substrate/frame/support/procedural/src/lib.rs

```diff
@@ -779,11 +779,20 @@ pub fn register_default_impl(attrs: TokenStream, tokens: TokenStream) -> TokenSt
 	}
 }
 
-///
-/// ---
-///
-/// Documentation for this macro can be found at
-/// `frame_support::pallet_prelude::inject_runtime_type`.
+/// The optional attribute `#[inject_runtime_type]` can be attached to `RuntimeCall`,
+/// `RuntimeEvent`, `RuntimeOrigin` or `PalletInfo` in an impl statement that has
+/// `#[register_default_impl]` attached to indicate that this item is generated by
+/// `construct_runtime`.
+///
+/// Attaching this attribute to such an item ensures that the combined impl generated via
+/// [`#[derive_impl(..)]`](macro@derive_impl) will use the correct type auto-generated by
+/// `construct_runtime!`.
+#[doc = docify::embed!("examples/inject_runtime_type.rs", derive_impl_works_with_runtime_type_injection)]
+///
+/// However, if `no_aggregated_types` is specified while using
+/// [`#[derive_impl(..)]`](macro@derive_impl), then these items are attached verbatim to the
+/// combined impl.
+#[doc = docify::embed!("examples/inject_runtime_type.rs", derive_impl_works_with_no_aggregated_types)]
```

Comment:
As a sidenote I find it a bit unfortunate that there is no button to click on to access the underlying file.

If there was a `source` button it would be great IMO.